### PR TITLE
Extract header validation logic into its own method

### PR DESCRIPTION
**Pull request by bachand**
In a previous PR we have discussed adding a public `validateHeader()` method: https://github.com/dfed/CacheAdvance/pull/22#discussion_r362097709

Right now if you instantiate a `CacheAdvance` object with metadata that doesn't match the metadata of an existing cache file, we reset that file.

Ideally we would have public API that allows you to first check if the file matches the metadata you are supplying, in order to prevent unexpected data loss.

This PR does not add a `validateHeader()` method. But by extracting the existing validation logic into its own method we're one step closer.

### CODE REVIEW

File: Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift [hunk 1]
```diff
@@ -85,6 +85,84 @@ final class CacheHeaderHandleTests: XCTestCase {
         XCTAssertEqual(headerHandle2.offsetInFileOfOldestMessage, defaultOffsetInFileOfOldestMessage)
     }
 
+    func test_synchronizeHeaderData_resetsFileWhenFileHeaderCannotBeCreated() throws {
```
Comment (bachand) [inline Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift]: This test exist to hit this code path, which we weren't hitting after my refactor https://github.com/dfed/CacheAdvance/blob/3ec3fab1d51f25ece5a2c1a0465b979ceb2e234d/Sources/CacheAdvance/CacheHeaderHandle.swift#L117

File: Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift [hunk 2]
```diff
@@ -85,6 +85,86 @@ final class CacheHeaderHandleTests: XCTestCase {
...
+        // Write a file that is too short for us to parse a `FileHeader` object.
+        let handle = try FileHandle(forUpdating: testFileLocation)
+        handle.write(Data(repeating: 0, count: Int(FileHeader.expectedEndOfHeaderInFile) - 1))
+
+        let headerHandle1 = try createHeaderHandle()
```
[the remark this answers is not in the document]
Comment (bachand) [reply Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift]: I will remove. I was messing around with various iterations of this test and forgot to simplify.

File: Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift [hunk 3]
```diff
@@ -85,6 +85,86 @@ final class CacheHeaderHandleTests: XCTestCase {
...
+        try headerHandle1.synchronizeHeaderData()
+
+        // Verify that the file is now the size of the header. This means that we rewrote the file.
```
[the remark this answers is not in the document]
Comment (bachand) [reply Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift]: `validateMetadata(against:)` takes as a parameter a `FileHeader` object that has been decoded from `Data`. Here we are seeing if it’s even possible to create a file header object. So by the time we get to `validateMetadata(against:)` we _know_ it’s possible, and its return value isn’t useful to this test. Here we are testing a code path outside of `validateMetadata(against:)`.

An alternative here would be trying to instantiate a `FileHeader` from `headerData`. We would assert that the file header is not `nil`. IIRC (currently on mobile) the first check in the `FileHeader` initializer is to verify the length of the data, like we’re doing here.

Let me know if you have a preference.

[the remark this answers is not in the document]
Comment (dfed) [reply Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift]: Aha! Okay this makes sense. I like that we're keeping the test simple as-is, and not jumping through more hoops than we need to. Thank you for explaining!

File: Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (bachand) [reply Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift]: 8209aebdb6110a9da2c488da987d0073e792f156

File: Sources/CacheAdvance/CacheHeaderHandle.swift [hunk 4]
```diff
@@ -72,6 +72,32 @@ final class CacheHeaderHandle {
         try handle.write(data: currentHeader.data(for: .offsetInFileAtEndOfNewestMessage))
     }
 
+    /// Checks if the static header metadata of this object matches another file header. Static metadata are the properties of the header
+    /// that need to remain the same for the entire lifetime of a cache file.
+    ///
+    /// - Parameter fileHeader: A representation of the header data in an existing cache file.
+    ///
+    /// - Returns: `true` if this object's static metadata matches that of `fileHeader`; otherwise `false`.
+    func validateMetadata(against fileHeader: FileHeader) -> Bool {
```
[the remark this answers is not in the document]
Comment (bachand) [reply Sources/CacheAdvance/CacheHeaderHandle.swift]: I used the name you suggested: 34c020665579493f914c8a110fd9cffb4243f1eb

**Pull request merged**